toolchain/check/handle_impl.cpp

```diff
@@ -37,25 +37,16 @@ auto HandleParseNode(Context& context, Parse::ImplIntroducerId node_id)
 
   // This might be a generic impl.
   StartGenericDecl(context);
+  return true;
+}
 
-  // Push a pattern block for the signature of the `forall` (if any).
-  // TODO: Instead use a separate parse node kinds for `impl` and `impl forall`,
-  // and only push a pattern block in `forall` case.
+auto HandleParseNode(Context& context, Parse::ForallId /*node_id*/) -> bool {
```

Comment:
Maybe retain some of the prior comment, as in:

```suggestion
auto HandleParseNode(Context& context, Parse::ForallId /*node_id*/) -> bool {
  // Push a pattern block for the signature of the `forall`.
```